include/envoy/network/transport_socket.h

```diff
@@ -144,7 +145,7 @@ typedef std::unique_ptr<TransportSocket> TransportSocketPtr;
 /**
  * A factory for creating transport socket. It will be associated to filter chains and clusters.
  */
-class TransportSocketFactory {
+class TransportSocketFactory : public Secret::SecretCallbacks {
```

Comment:
revert this file, not all TransportSocketFacotry need this, it can be done in impl side.

Reply:
Reverted